packages/flame_jenny/jenny/lib/src/dialogue_view.dart

```diff
@@ -7,13 +7,35 @@ import 'package:jenny/src/structure/dialogue_choice.dart';
 import 'package:jenny/src/structure/dialogue_line.dart';
 import 'package:jenny/src/structure/dialogue_option.dart';
 import 'package:jenny/src/structure/node.dart';
+import 'package:jenny/src/yarn_project.dart';
 import 'package:meta/meta.dart';
 
+/// The **DialogueView** class is the main mechanism for integrating Jenny with
+/// a game engine. This class describes how {ref}`line <DialogueLine>`s and
+/// {ref}`option <DialogueOption>`s are presented to the user.
+///
+/// The class is abstract, which means you must create a concrete
+/// implementation in order to use Jenny's dialogue system. The concrete
+/// `DialogueView` objects will then be passed to a [DialogueRunner], which
+/// will orchestrate the dialogue's progression.
+///
+/// The class defines a number of "event handler" methods, which can be
+/// overridden in subclasses in order to respond to the corresponding event.
+/// Each method has a default no-op implementation, which means you only need
+/// to override those methods that you care about.
+///
+/// Most of the event handler methods return [FutureOr], which means they can
+/// be implemented either synchronously or asynchronously. In the latter case
+/// the dialogue runner will wait for the future to resolve before proceeding
+/// (futures from several dialogue views will be awaited simultaneously).
```

Comment:
Were these Jenny fixes supposed to be in this PR? 🤔

Reply:
Yes; the changes in Jenny are only related to doc-comments. I needed to test the new Dart domain feature on something, and Jenny classes seemed like a good fit. The end result is that some content from the docs site ends up being transferred into the code. Which is good, because it means the docs can now only be written in one place, and don't have to be duplicated.

So what you see as changes in the Jenny code, is really just me combining docs with the code doc-comments.

Reply:
Sounds good, didn't read it that carefully on the first pass.